src/engine/Test_engine.ml

```diff
@@ -325,7 +326,8 @@ let make_tests ?(unit_testing = false) ?(get_xlang = None)
   let print_summary () =
```

Comment:
maybe print_summary_and_exit_if_errors?
And change also the caller?

exit is pretty radical, so I usually prefer to know if a function can potentially exit.

Reply:
True, I instead made it to return `!total_mismatch` and the `exit` I moved it to `test_rules`.